overlord/servicestate/quota_control.go

```diff
@@ -58,7 +67,59 @@ func EnsureQuotaUsability() (restore func()) {
 	}
 }
 
+func checkMemoryCGroupEnabled() {
+	memoryCGroupError = memoryCGroupEnabled()
+}
+
+// added to enable path update for testing purposes
+func SetCGroupsFilePath(path string) {
+	CGroupsFilePath = path
+	checkMemoryCGroupEnabled()
+}
+
+// since the control groups can be enabled/disabled without the kernel config the only
+// way to identify the status of memory control groups is via /proc/cgroups
+// "cat /proc/cgroups | grep memory" returns the active status of memory control group
+// and the 3rd parameter is the status
+// 0 => false => disabled
+// 1 => true => enabled
+func memoryCGroupEnabled() error {
```

Comment:
super nitpick: I would expect a function with this name to return a simple `bool`; as it seems that the whole point of this PR is to retrieve a meaningful error message and returning a bool is not an option, I'd vote to rename this to `ensureMemoryCGroupEnabled()`, because then it would be clear that this would return an error when the condition is not fulfilled.

Reply:
I think the presence of "Enabled" in the end may be the culprit here, how about memoryCGroupCheck instead?